diff -rudN change-03-threshold_default_75/src/commandline.cc change-04-add_timecode_to_graph/src/commandline.cc
--- change-03-threshold_default_75/src/commandline.cc	2013-08-12 01:16:16.129550595 +0200
+++ change-04-add_timecode_to_graph/src/commandline.cc	2013-08-12 01:10:02.829556995 +0200
@@ -48,6 +48,7 @@
   printf ("\nShotdetect version \"%s\", Copyright (c) 2007-2013 Johan Mathe\n\n"
           "Usage: %s \n"
           "-h           : show this help\n"
+          "-n           : commandline mode (disable GUI)\n"
           "-s threshold : threshold (Default=%d)\n"
           "-i file      : input file path\n"
           "-o path      : output path\n"
@@ -59,7 +60,8 @@
           "-f           : generate first image for each shot\n"
           "-l           : generate last image for each shot\n"
           "-m           : generate the thumbnail image\n"
-          "-r           : generate the images in native resolution\n",
+          "-r           : generate the images in native resolution\n"
+          "-c           : print timecode on x-axis in graph\n",
           APP_VERSION,
           argv[0],
           DEFAULT_THRESHOLD
@@ -87,7 +89,7 @@
   f.threshold=DEFAULT_THRESHOLD;
 
   for (;;) {
-    int c = getopt (argc, argv, "?ht:y:i:o:a:x:s:flwvmr");
+    int c = getopt (argc, argv, "?ht:y:i:o:a:x:s:flwvmrc");
 
     if (c < 0) {
       break;
@@ -135,6 +137,11 @@
       f.set_threshold(atoi (optarg));
       break;
 
+      /* Embed timecode in graph  */
+    case 'c':
+      f.set_show_timecode(true);
+      break;
+
       /* Id for path creation */
     case 'a':
       f.set_alphaid (optarg);
diff -rudN change-03-threshold_default_75/src/film.h change-04-add_timecode_to_graph/src/film.h
--- change-03-threshold_default_75/src/film.h	2013-08-12 01:01:35.869565681 +0200
+++ change-04-add_timecode_to_graph/src/film.h	2013-08-12 01:08:40.265558408 +0200
@@ -179,6 +179,8 @@
   string code_lang;
   /* Processing threshold */
   int threshold;
+  /* Embed timecode */
+  int show_timecode;
   /* Alphanumeric ID */
   string alphaid;
   /* Absolute path */
@@ -226,6 +228,9 @@
   inline void set_threshold(int threshold) {
     this->threshold = threshold;
   };
+  inline void set_show_timecode(bool val) {
+    this->show_timecode = val;
+  };
   inline void set_ipath(string path) {
     this->input_path = path;
   };
diff -rudN change-03-threshold_default_75/src/graph.cc change-04-add_timecode_to_graph/src/graph.cc
--- change-03-threshold_default_75/src/graph.cc	2013-08-12 00:30:10.385597993 +0200
+++ change-04-add_timecode_to_graph/src/graph.cc	2013-08-12 01:07:51.721559240 +0200
@@ -100,6 +100,7 @@
   title_color = gdImageColorAllocate (im_motion_qty, 255, 255, 0);
   grid_color = gdImageColorAllocate (im_motion_qty, 0, 0, 0);
   threshold_color = gdImageColorAllocate (im_motion_qty, 255, 0, 0);
+  timecode_color = grid_color;
 
 
   /*
@@ -228,15 +229,17 @@
 
         // Write video position as grid label:
         //printf("%d:%02d:%02d.%02d\n", hour, minute, second, frame);
-        /*sprintf(time, "%d:%02d:%02d.%02d", hour, minute, second, frame);
+        if (this->f->show_timecode == true) {
+        sprintf(time, "%d:%02d:%02d.%02d", hour, minute, second, frame);
 
         gdImageString (im,
           gdFontGetLarge(),
           x - (strlen(time) * gdFontGetLarge()->w / 2),
           xaxis_offset + 8,
           (unsigned char *) time,
-          grid_color
-        );*/
+          timecode_color
+        );
+        }
       }
       //if (!(i % grid_size)) //CF
       gdImageLine (im, x, xaxis_offset - 1, x, xaxis_offset + tick_length, grid_color);
diff -rudN change-03-threshold_default_75/src/graph.h change-04-add_timecode_to_graph/src/graph.h
--- change-03-threshold_default_75/src/graph.h	2013-08-11 22:45:59.453705117 +0200
+++ change-04-add_timecode_to_graph/src/graph.h	2013-08-12 01:08:03.689559036 +0200
@@ -94,6 +94,7 @@
   int green_color;
   int blue_color;
   int threshold_color;
+  int timecode_color;
   void draw_canvas (gdImagePtr im, string title);
   float MAX(float a, float b, float c);
   float MIN(float a, float b, float c);
diff -rudN change-03-threshold_default_75/src/main.cc change-04-add_timecode_to_graph/src/main.cc
--- change-03-threshold_default_75/src/main.cc	2013-08-12 00:59:59.021567341 +0200
+++ change-04-add_timecode_to_graph/src/main.cc	2013-08-12 01:07:23.757559721 +0200
@@ -56,7 +56,8 @@
           "-f           : generate first image for each shot\n"
           "-l           : generate last image for each shot\n"
           "-m           : generate the thumbnail image\n"
-          "-r           : generate the images in native resolution\n",
+          "-r           : generate the images in native resolution\n"
+          "-c           : print timecode on x-axis in graph\n",
           APP_VERSION,
           argv[0],
           DEFAULT_THRESHOLD
@@ -82,7 +83,7 @@
   f.threshold=DEFAULT_THRESHOLD;
 
   for (;;) {
-    int c = getopt (argc, argv, "?hnt:y:i:o:a:x:s:flwvmr");
+    int c = getopt (argc, argv, "?hnt:y:i:o:a:x:s:flwvmrc");
 
     if (c < 0) {
       break;
@@ -130,6 +131,10 @@
       f.set_threshold(atoi (optarg));
       break;
 
+      /* Embed timecode in graph  */
+    case 't':
+      f.set_show_timecode(true);
+      break;
 
       /* Id for path creation */
     case 'a':
